### PR TITLE
Use external Vault

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

v2 version of https://github.com/kuzzleio/kuzzle/pull/1508

Enables Kuzzle Core to use the external vault. This solves the problem that Kuzzle was not able to decrypt a secrets file previously encrypted via the CLI (which uses the new, externalized, vault).

### How should this be manually tested?

1. Create a secrets file using the CLI.
2. Start a new Kuzzle instance using this secrets file.
3. Verify that Kuzzle starts up correctly.

### Other changes

 - Load secrets from the vault for test and dev environments
 - fix a bug on the `kuzzlerc.default` file

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/6b42fc1c8f817557800922f10661073f1be89f8d/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/83f7b4d63bc4526c1c6313ba18a2c3e4d81a4629/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/776646b54db5b26a5ba62c907a057f318eb491d4/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/b0f835b06ab16126463f2f99afd317c21d6f567c/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/df6128b134ab594a4d25d51610afb69112e280a3/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/350fe1261befd1f60b8040ce1340586520faaffb/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: docker-compose.yml [hunk 1]
```diff
@@ -1,54 +1,54 @@
...
 services:
-  nginx:
-    image: nginx:1.15-alpine
-    depends_on:
-      - kuzzle
-    ports:
-      - "7443:7443"
-    volumes:
-      - ./docker-compose/nginx:/etc/nginx/conf.d
+  # nginx:
```
Comment (scottinet) [inline docker-compose.yml]: debug leftover?

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/1f15b1024c40a930bebd051560c0c272ae56faf4/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: docker-compose.yml [hunk 2]
```diff
@@ -49,6 +49,8 @@ services:
       - KUZZLE_VAULT_KEY=${KUZZLE_VAULT_KEY}
       - NODE_10_VERSION=10.16.0
       - WITHOUT_KUZZLE=${WITHOUT_KUZZLE}
+      - KUZZLE_SECRETS_FILE=/var/app/features-sdk/fixtures/secrets.enc.json
+      - KUZZLE_VAULT_KEY=secret-password
```
Comment (Yoann-Abbes) [inline docker-compose.yml]: Why is there KUZZLE_VAULT_KEY defined twice ?

Comment (Aschen) [reply docker-compose.yml]: Good catch :sweat_smile:

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/46c15be8e34541ac45bbd3c060c963e14d0dec6b/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/2c5e9b44b0897cc21ba1a7f9d0bc14e8973816c3/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/48a417161feb36d5851f27eb8680353c63867f27/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [kuzzle.js#L55](https://github.com/kuzzleio/kuzzle/blob/1a348436d6a5b1bd56d9833e33c2a7c79ee4e36e/lib/api/kuzzle.js#L55): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**